pennylane/compiler/compiler.py

```diff
@@ -18,6 +18,11 @@
 from importlib import reload, metadata
 from collections import defaultdict
 import dataclasses
+import os
+
+from packaging import version
+
+MIN_CATALYST_VERSION = version.parse(os.environ.get("MIN_CATALYST_VERSION", "0.4.0"))
```

Comment:
```suggestion
from semantic_version import Version

MIN_CATALYST_VERSION = Version(os.environ.get("MIN_CATALYST_VERSION", "0.4.0"))
```

Reply:
thanks @timmysilv :+1:

Reply:
nice, I discussed this with Lee in advanced about common practices here. I wasn't aware of `semantic_version` :+1: